src/dehydrated_devices.rs

```diff
@@ -33,18 +78,54 @@ impl DehydratedDevices {
     #[wasm_bindgen]
     pub async fn rehydrate(
         &self,
-        pickle_key: &Uint8Array,
+        pickle_key: &DehydratedDeviceKey,
         device_id: &DeviceId,
         device_data: &str,
     ) -> Result<RehydratedDevice, JsError> {
-        let pickle_key: [u8; 32] =
-            pickle_key.to_vec().try_into().map_err(|_| JsError::new("Wrong key length"))?;
+        let sdk_pickle_key =
+            SdkDehydratedDeviceKey::from_slice(pickle_key.inner.to_vec().as_slice())?;
+
         Ok(self
             .inner
-            .rehydrate(&pickle_key, &device_id.inner, serde_json::from_str(device_data)?)
+            .rehydrate(&sdk_pickle_key, &device_id.inner, serde_json::from_str(device_data)?)
             .await?
             .into())
     }
+
+    /// Get the cached dehydrated device pickle key if any.
+    ///
+    /// None if the key was not previously cached (via
+    /// [`Self::save_dehydrated_device_pickle_key`]).
+    ///
+    /// Should be used to periodically rotate the dehydrated device to avoid
+    /// OTK exhaustion and accumulation of to_device messages.
+    #[wasm_bindgen(js_name = "getDehydratedDeviceKey")]
+    pub async fn get_dehydrated_device_key(&self) -> Result<Option<DehydratedDeviceKey>, JsError> {
+        let key = self.inner.get_dehydrated_device_pickle_key().await?;
+        Ok(key.map(DehydratedDeviceKey::from))
```

Comment:
maybe simpler:

```suggestion
        Ok(key.into())
```

Reply:
I don't have a From<Option<>> , not sure to get it?

Reply:
sorry, you're right, this doesn't work.